lib/puppet/pops/parser/heredoc_support.rb

```diff
@@ -64,7 +64,7 @@ def heredoc
     # and it should start scanning after the first found \n (or if not found == error).
 
     if ctx[:newline_jump]
-      scn.pos = lexing_context[:newline_jump]
+      scn.pos = @lexing_context[:newline_jump]
```

Comment:
it should actually be `ctx[:newline_jump]` because it is much faster to access the local variable than the instance variable. At one point the code was refactored to use the shorter `ctx` instead of `lexing_context`. Apparently, this location was missed.